js2-mode.el

```diff
@@ -11272,6 +11283,204 @@ Selecting an error will jump it to the corresponding source-buffer error.
+          (>= current-line tag-start-line)))
+       (cond
+        ;; Analyze bounds if there are any
+        ((progn
+           (while (and (not tag-end-pos)
+                       (setq last-pos (re-search-forward js2-jsx-end-tag-re nil t)))
+             (goto-char (match-end 0))
+             (js2-forward-sws)
+             (when (looking-at js2-jsx-after-tag-re)
+               (setq tag-end-pos last-pos
+                     after-tag-pos (match-beginning 0))))
+           tag-end-pos)
+         (setq tag-end-line (line-number-at-pos tag-end-pos)
+               after-tag-line (line-number-at-pos after-tag-line))
+         (or (and
+              ;; Ensure we're actually within the bounds of the jsx
+              (<= current-line tag-end-line)
+              ;; An "after" line which does not end an element begins with
+              ;; js, so indent it like js
+              (<= current-line after-tag-line))
+             (and
+              ;; Handle another case where there could be e.g. comments after
+              ;; the element
+              (> current-line tag-end-line)
+              (< current-line after-tag-line)
+              (setq type 'after))))
+        ;; They may not be any bounds (yet)
+        (t))
+       ;; Check if we're inside an embedded multi-line js expression
+       (cond
+        ((not type)
+         (goto-char current-pos)
+         (end-of-line)
+         (setq parens (nth 9 (syntax-ppss)))
+         (while (and parens (not type))
+           (setq paren (car parens))
+           (cond
+            ((and (>= paren tag-start-pos)
+                  ;; Curly bracket indicates the start of an embedded expression
+                  (= (char-after paren) 123) ; {
+                  ;; The first line of the expression is indented like sgml
+                  (> current-line (line-number-at-pos paren))
+                  ;; Check if within a closing curly bracket (if any)
+                  ;; (exclusive, as the closing bracket is indented like sgml)
+                  (cond
+                   ((progn
+                      (goto-char paren)
+                      (ignore-errors (let (forward-sexp-function)
+                                       (forward-sexp))))
+                    (< current-line (line-number-at-pos)))
+                   (t)))
+             ;; Indicate this guy will be indented specially
+             (setq type 'expression))
+            (t (setq parens (cdr parens)))))
+         t)
+        (t))
+       (cond
+        (type)
+        ;; Indent the first jsx thing like js so we can indent future jsx things
+        ;; like sgml relative to the first thing
+        ((= current-line tag-start-line) 'first)
+        ('nth))))))
+
+(defmacro js2-as-sgml (&rest body)
+  "Execute BODY as if in sgml-mode."
+  `(with-syntax-table sgml-mode-syntax-table
+     (let (forward-sexp-function
+           parse-sexp-lookup-properties
+           (sgml-basic-offset js2-basic-offset))
```

Comment:
Pretty sure. JSX is described as an extension of JS, rather than as "embedded XML in JS". In most code snippets I've seen, the indentation people use is the same. Due to many editors binding TAB to "insert N whitespace characters," it's likely that other people on a team will use consistent indentation throughout JSX files.

Although, `sgml-attribute-offset` is not also rebound, and that could be anything.

So I see a few options here:
- Make a `js2-jsx-basic-offset` and/or `js2-jsx-attribute-offset`
  - Pros: Neither standard JS nor SGML preferences necessarily dictate this indentation
  - Cons: More variables, and the user likely(?) wants the offset to be the same
- Use `sgml-basic-offset` and `sgml-attribute-offset`
  - Pros: Fewer variables
  - Cons: Presumes that SGML preferences carry over to JS/JSX